python/packages/autogen-ext/src/autogen_ext/tools/azure/_ai_search.py

```diff
@@ -0,0 +1,1616 @@
+        """Factory method to create a full text search tool.
+
+        Full text search provides enhanced text analysis with Azure's advanced text
+        processing capabilities, including word breaking, stemming, and relevance scoring.
+        This provides more sophisticated keyword matching than simple search.
+
+        Args:
+            name (str): The name of the tool
+            endpoint (str): The URL of your Azure AI Search service
+            index_name (str): The name of the search index
+            credential (Union[AzureKeyCredential, TokenCredential, Dict[str, str]]): Authentication credentials
+            search_fields (Optional[List[str]]): Optional list of fields to search within
+            select_fields (Optional[List[str]]): Optional list of fields to return in the results
+            **kwargs (Any): Additional configuration options
+
+        Returns:
+            An initialized full text search tool
+
+        Example:
+            .. code-block:: python
+
+                from azure.core.credentials import AzureKeyCredential
+                from autogen_ext.tools.azure import AzureAISearchTool
+
+                # Create a full text search tool
+                full_search = AzureAISearchTool.create_full_search(
+                    name="full_search",
+                    endpoint="https://your-search-service.search.windows.net",
+                    index_name="your-index",
+                    credential=AzureKeyCredential("your-api-key"),
+                    search_fields=["title", "content"],
+                    select_fields=["id", "title", "content", "url"],
+                )
+
+                # Perform a full text search
+                results = await full_search.run(args={"query": "advanced neural network architectures"})
+
+                # Process results
+                for item in results.results:
+                    print(f"Score: {item.score}, Title: {item.content.get('title', 'No title')}")
+        """
+        if not endpoint or not endpoint.startswith(("http://", "https://")):
+            raise ValueError("endpoint must be a valid URL starting with http:// or https://")
+
+        if not index_name:
+            raise ValueError("index_name cannot be empty")
+
+        if not name:
+            raise ValueError("name cannot be empty")
+
+        if not credential:
+            raise ValueError("credential cannot be None")
+
+        token = _allow_private_constructor.set(True)
+        try:
+            return cls(
+                name=name,
+                endpoint=endpoint,
+                index_name=index_name,
+                credential=credential,
+                query_type="full",
+                search_fields=search_fields,
+                select_fields=select_fields,
+                **kwargs,
+            )
+        finally:
+            _allow_private_constructor.reset(token)
+
+    @classmethod
+    def create_semantic_search(
+        cls,
```

Comment:
I don't see "semantic search" as a query type in Azure AI Search: https://learn.microsoft.com/en-us/azure/search/search-query-overview